doc/rados/operations/placement-groups.rst

```diff
@@ -390,172 +407,162 @@ group will store objects on two OSDs, as shown below.
+When the number of PGs increases, several consequences ensue. The new PGs are
+assigned OSDs.  The result of the CRUSH function changes, which means that some
+objects from the already-existing PGs are copied to the new PGs and removed
+from the old ones.
 
-Placement Groups Tradeoffs
-==========================
+Factors Relevant To Specifying pg_num
+=====================================
 
-Data durability and even distribution among all OSDs call for more
-placement groups but their number should be reduced to the minimum to
-save CPU and memory.
+On the one hand, the criteria of data durability and even distribution across
+OSDs weigh in favor of a high number of PGs. On the other hand, the criteria of
+saving CPU resources and minimizing memory usage weigh in favor of a low number
+of PGs.
 
 .. _data durability:
 
 Data durability
 ---------------
 
-After an OSD fails, the risk of data loss increases until the data it
-contained is fully recovered. Let's imagine a scenario that causes
-permanent data loss in a single placement group:
-
-- The OSD fails and all copies of the object it contains are lost.
-  For all objects within the placement group the number of replica
-  suddenly drops from three to two.
-
-- Ceph starts recovery for this placement group by choosing a new OSD
-  to re-create the third copy of all objects.
-
-- Another OSD, within the same placement group, fails before the new
-  OSD is fully populated with the third copy. Some objects will then
-  only have one surviving copies.
-
-- Ceph picks yet another OSD and keeps copying objects to restore the
-  desired number of copies.
-
-- A third OSD, within the same placement group, fails before recovery
-  is complete. If this OSD contained the only remaining copy of an
-  object, it is permanently lost.
-
-In a cluster containing 10 OSDs with 512 placement groups in a three
-replica pool, CRUSH will give each placement groups three OSDs. In the
-end, each OSDs will end up hosting (512 * 3) / 10 = ~150 Placement
-Groups. When the first OSD fails, the above scenario will therefore
-start recovery for all 150 placement groups at the same time.
-
-The 150 placement groups being recovered are likely to be
-homogeneously spread over the 9 remaining OSDs. Each remaining OSD is
-therefore likely to send copies of objects to all others and also
-receive some new objects to be stored because they became part of a
-new placement group.
-
-The amount of time it takes for this recovery to complete entirely
-depends on the architecture of the Ceph cluster. Let say each OSD is
-hosted by a 1TB SSD on a single machine and all of them are connected
-to a 10Gb/s switch and the recovery for a single OSD completes within
-M minutes. If there are two OSDs per machine using spinners with no
-SSD journal and a 1Gb/s switch, it will at least be an order of
+After an OSD fails, the risk of data loss continues to increase until the data
```

Comment:
When an OSD fails, the risk of data loss is increased until replication of the data it hosted is restored to the configured level.

Reply:
Much better. Thank you.

Accepted. Obviously.